6RunTumbleTooSlow.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Dec  6 16:12:29 2018
+
+@author: JINGYUE 
+
+This code has not been optimised
+"""
+import matplotlib.pyplot as plt  
+import random
+import math 
+import numpy as np
+import time
+import scipy.constants as sc
+
+# to start the timer:
+start_time = time.time()  
+
+# the constants:
+hotIndex = 820 # the chosen index number of the hot particle os that it's near the centre of the plot
+d = 86  # dimension of the square box, to ensure 0.68 area fraction
+n = 1600  # number of particles 
+a = 1 # radius of the particles 
+D = 0.1 # the diffusion constant
+T = 298 # the passive medium temperature
+u = (sc.k)*T*10**13 # the mobility of the particle， from the formula that u = Kb*T/D
+k = 100*D/(u*(a**2)) # the spring constant 
+vo = 100 # the self propulsion velocity, 100
+c = 4430 #### the number of timesteps iterated, by right, it's 2,000,000 = 10 time units
+t = 0.5*(10**(-5)) #### **(-5) unit length of the timestep/time interval
+tao = 0.00025 # tumbling coefficient, when it's 1, the active particle will produce an empty streak. When it's 0.01 or below, a void is expected            
+rott = tao*(a**2)*D  # the time interval between tumblings, equal to 2.5e-5
+m = rott/t  # number of frames between each tumling event
+l = (d - 2*a*math.sqrt(n))/math.sqrt(n)  # the width of the fringe around each particle x2
+it0 = 10   # it0, t0max, and tmax are related to the diffuse() function
+t0max = int((c - it0)/it0)
+tmax = c - it0
+z = 1000  # the plot() function is called every z timesteps
+
+# the variables:
+angle = 0 # random.uniform(-math.pi, math.pi)
+_ = 0 # just a frame counter
+xy = np.empty([n, 2])  # an empty array xy that will store elements of the form [x, y]
+i = (2*a + l) / 2   # the start of the lattice distribution
+p = 0
+
+# to first distribute the particles in a semi-random square lattice
+while i < d:  
+    j = (2*a + l) / 2

```

Comment:
what is j? and what is cur? i think the variable names can be more descriptive

Reply:
i and j are the reference positions along rows and columns. 
cur is the initial position chosen for a particle, after adjusting the values of i and j with a bit of random shift. 
they are good enough as names for now. I will make them more descriptive when it is fully ready. 
thanks!